### DISCUSSION TITLE
Convert `Gmpq` number to string

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#6607] [Ideas]
Hello,

  I needed to convert a `Gmpq` rational number to a string and I didn't find a built-in way in the doc (nor in the source code). So I wrote my own function (it requires `gmp.h`):

```cpp
std::string q2str(CGAL::Gmpq r){
  CGAL::Gmpz numer = r.numerator();
  CGAL::Gmpz denom = r.denominator();
  size_t n = mpz_sizeinbase(numer.mpz(), 10) + 2;
  size_t d = mpz_sizeinbase(denom.mpz(), 10) + 2;
  char *cnumer = new char [n];
  char *cdenom = new char [d];
  cnumer = mpz_get_str(cnumer, 10, numer.mpz());
  cdenom = mpz_get_str(cdenom, 10, denom.mpz());
  std::string snumer = cnumer;
  std::string sdenom = cdenom;
  delete[] cnumer;
  delete[] cdenom;
  return snumer + "/" + sdenom;
}
```

I think that would be nice to have this feature in CGAL.

By the way, unless I missed it, `Gmpq::to_double` is not documented.

**Comment by pentacular** [comment]
Doesn't this do what you want?
```
  std::ostringstream x;
  x << centroid.x().exact();
  std::string xs = x.str();
```

Also, please see CGAL::to_double()

**Comment by stla** [comment]
Thanks. That's fine if this works with `Gmpq`. I would only complain that it looks a bit tortuous. A `Gmpq::to_string` would be more friendly.

That's also fine if `CGAL::to_double()` works with `Gmpq`, but it's strange to document `Gmpz::to_double` and not `Gmpq::to_double`.

**Reply by afabri** [reply]
It is a free function as one can then also write for any third party number type library where we could not  add a function inside a class.
